### PR TITLE
DEV: add system test for insert-video modal and video output

### PR DESCRIPTION
**Pull request by tyb-talks**
This makes the insert-video feature more resilient to regressions, will support an upcoming modal upgrade.

Manual test to verify video playback works after system test is run:

https://github.com/discourse/discourse-insert-video/assets/133760061/4e259412-c9b3-4a9c-a539-9b5c9a8eddba

### CODE REVIEW

**Comment by tgxworld** [comment]
@tyb-talks is it possible to use a generic video and image for the test here to eliminate any risk of us distributing content which we do not own the license to.

**Comment by martin-brennan** [comment]
@tgxworld Big Buck Bunny has been used for years for this kind of thing (not just Discourse), it's an open source film made by blender released under a creative commons license. https://www.protocol.com/big-buck-bunny-video-industry

**Comment by tgxworld** [comment]
O icic. That is good to know. Sorry about the distraction here @tyb-talks

File: spec/system/insert_video_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        Pathname.new("#{__FILE__}/../../fixtures/media/sample_video.mp4")
+      )
+    poster_file =
+      File.absolute_path(
+        Pathname.new("#{__FILE__}/../../fixtures/images/poster_small.jpg")
+      )
+    subtitle_file =
+      File.absolute_path(
+        Pathname.new("#{__FILE__}/../../fixtures/media/sample_video.vtt")
+      )
```
[the remark this answers is not in the document]
Comment (tgxworld) [reply spec/system/insert_video_spec.rb:26]: I agree this is fine here :+1:

File: spec/system/page_objects/modals/insert_video.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      BODY_SELECTOR = ".insert-video"
+      MODAL_SELECTOR = ".insert-video-modal"
+      def click_add_video_source_button
+        find(".video-sources .simple-list-uploader .btn").click
+      end
+
+      def click_add_poster_button
+        find(".video-poster .simple-list-uploader .btn").click
+      end
+      def click_add_subtitle_button
```
Comment (tgxworld) [inline spec/system/page_objects/modals/insert_video.rb:17]:
```suggestion
      end

      def click_add_subtitle_button
```

File: spec/system/page_objects/modals/insert_video.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        find(".video-subtitles .simple-list-uploader .btn").click
+      end
+
+      def click_primary_button
+        find(".modal-footer .btn-primary").click
```
Comment (tgxworld) [inline spec/system/page_objects/modals/insert_video.rb:22]: This is a little too generic IMO. The test may start to fail if there are multiple primary buttons in the future.

**Review by tgxworld — APPROVED** [review]

File: spec/system/page_objects/modals/insert_video.rb (hunk 3 above)
Comment (martin-brennan) [reply spec/system/page_objects/modals/insert_video.rb:22]: @tgxworld and @tyb-talks this `click_primary_button` code is already in the base modal page object class too, so no need to reproduce here:

https://github.com/discourse/discourse/blob/c197f8c6ee59f56ba9f830388a36ecfd97b7cbd5/spec/system/page_objects/modals/base.rb#L31-L33

I think it's okay for now, but yes we should probably have a more generic `click_footer_button` for the modal base.

As for d-button, I can't find any places where that uses an ID in core either :( So I think the best you can do is adding a unique CSS class, I think that would be best, something like `btn-insert-video__submit`. There should only ever be 1 cancel button so you can rely on Base for that:

https://github.com/discourse/discourse/blob/c197f8c6ee59f56ba9f830388a36ecfd97b7cbd5/spec/system/page_objects/modals/base.rb#L23-L25

**Pull request merged**